dtfs-central-api/src/helpers/get-previous-report-period.test.ts

```diff
@@ -37,19 +37,49 @@ describe('getPreviousReportPeriod', () => {
     });
   });
 
-  describe('when a bank is found', () => {
+  describe('when a bank is found and the month has 1 digit', () => {
     beforeEach(() => {
       jest.mocked(getBankById).mockImplementation(jest.fn().mockResolvedValueOnce(mockGetBankByIdResponse));
     });
 
     it('should return the result of getPreviousReportPeriodForBankScheduleByMonth', async () => {
+      const reportPeriod = { start: { month: 2, year: 2024 }, end: { month: 4, year: 2024 } };
+      utilisationReport = UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').withReportPeriod(reportPeriod).build();
       const response = await getPreviousReportPeriod(bankId, utilisationReport);
 
-      const { start } = utilisationReport.reportPeriod;
-      const paddedMonth = String(start.month).padStart(2, '0');
-      const monthFormatted = `${start.year}-${paddedMonth}`;
+      const expected = getPreviousReportPeriodForBankScheduleByMonth(mockGetBankByIdResponse.utilisationReportPeriodSchedule, '2024-02');
 
-      const expected = getPreviousReportPeriodForBankScheduleByMonth(mockGetBankByIdResponse.utilisationReportPeriodSchedule, monthFormatted);
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe('when a bank is found and the month has 2 digits', () => {
+    beforeEach(() => {
+      jest.mocked(getBankById).mockImplementation(jest.fn().mockResolvedValueOnce(mockGetBankByIdResponse));
+    });
+
+    it('should return the result of getPreviousReportPeriodForBankScheduleByMonth', async () => {
+      const reportPeriod = { start: { month: 10, year: 2024 }, end: { month: 4, year: 2024 } };
+      utilisationReport = UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION').withReportPeriod(reportPeriod).build();
+      const response = await getPreviousReportPeriod(bankId, utilisationReport);
+
+      const expected = getPreviousReportPeriodForBankScheduleByMonth(mockGetBankByIdResponse.utilisationReportPeriodSchedule, '2024-10');
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe('when a bank is found and the month has 2 digits and starts with 0', () => {
```

Comment:
```suggestion
  describe('when a bank is found and the start month of the report period is a single digit', () => {
```

Reply:
this is testing it works when the date is 03 format

Reply:
it is doing 03 basically (as it does not accept 03 written like that).  It probably would never but its a nice test just to make sure it works

Reply:
I'll approve since this is low stakes but I disagree, I believe this is an octal number and I don't see the point in using an octal number when we are never using octal numbers for report periods